src/index.js

```diff
@@ -143,6 +104,36 @@ export function useVirtual({
     return virtualItems
   }, [startIndex, endIndex, measurements, sizeKey, defaultScrollToFn])
 
+  /*
+  // FIXME: setRange() doesn't seem to trigger render
+  useScroll(parentRef, ({ [scrollKey]: scrollOffset }) => {
+    latestRef.current.scrollOffset = scrollOffset
+    setRange(prevRange => calculateRange(latestRef.current, prevRange))
+  })
+  */
+
+  ///*
+  // FIXME: Had to Copy the logic from useScroll to get setRange() to trigger render
+  React.useEffect(() => {
+    const element = parentRef.current
+
+    const handler = (event) => {
+      const scrollOffset = event.target[scrollKey]
+      latestRef.current.scrollOffset = scrollOffset
+      setRange(prevRange => calculateRange(latestRef.current, prevRange))
+    }
+
+    element.addEventListener('scroll', handler, {
+      capture: false,
+      passive: true,
+    })
+
+    return () => {
+      element.removeEventListener('scroll', handler)
+    }
+  }, [parentRef, scrollKey])
+  //*/
```

Comment:
For some reason I couldn't get the hook to work with the existing `useScroll()`. Sometimes the `setRange()` callback would only be called on the initial mount and not during scrolling. Any ideas on how to fix this?

```suggestion
  useScroll(parentRef, ({ [scrollKey]: scrollOffset }) => {
    latestRef.current.scrollOffset = scrollOffset
    setRange(prevRange => {
      console.log('setRange') // FIXME: Sometimes never triggers after the initial mount
      return calculateRange(latestRef.current, prevRange)
    })
  })
```